lib/logstash/inputs/beats.rb

```diff
@@ -189,6 +206,15 @@ def register
       @logger.warn("configured ssl_key => #{@ssl_key.inspect} will not be used") if @ssl_key
     end
 
+    @enrich = resolve_enriches
+
+    @include_codec_tag = original_params.include?('include_codec_tag') ? params['include_codec_tag'] : include_codec_metadata?
+    @ssl_peer_metadata = original_params.include?('ssl_peer_metadata') ? params['ssl_peer_metadata'] : include_source_metadata?
```

Comment:
we are considering `ssl_peer_metadata` in the default (through alias) enriches and applying `source_metadata` makes `ssl_peer_metadata` true if deprecated config is not provided (and we should avoid aligning on deprecated config)

```suggestion
    @ssl_peer_metadata = original_params.include?('ssl_peer_metadata') ? params['ssl_peer_metadata'] : @enrich.include?('ssl_peer_metadata')
```